client/a8c-for-agencies/sections/sites/site-actions/remove-site-action-modal.tsx

```diff
@@ -0,0 +1,84 @@
+import { Button, __experimentalHStack as HStack } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import useRemoveSiteMutation from 'calypso/a8c-for-agencies/data/sites/use-remove-site';
+import { useDispatch } from 'calypso/state';
+import { successNotice } from 'calypso/state/notices/actions';
+import type { SiteData } from '../../../../jetpack-cloud/sections/agency-dashboard/sites-overview/types';
+
+const createRemoveSiteActionModal =
+	( onRefetchSite: () => Promise< unknown > ) =>
+	( {
+		items,
+		closeModal,
+		onActionPerformed,
+	}: {
+		items: SiteData[];
+		closeModal: () => void;
+		onActionPerformed?: () => void;
+	} ) => {
+		const translate = useTranslate();
```

Comment:
This implements the same logic as [SiteRemoveConfirmationDialog](https://github.com/Automattic/wp-calypso/blob/add/actions-to-a4a-sites/client/a8c-for-agencies/sections/sites/site-remove-confirmation-dialog/index.tsx#L11) and sub-components.